src/Sil/Component/Invoice/Model/Invoice.php

```diff
@@ -0,0 +1,590 @@
+     * delivery date.
+     *
+     * @var DateTime
+     */
+    protected $date;
+
+    /**
+     * currency.
+     *
+     * @var CurrencyInterface
+     */
+    protected $currency;
+
+    /**
+     * status.
+     *
+     * @var InvoiceStatus
+     */
+    protected $status;
+
+    /**
+     * tax rate.
+     *
+     * @var float
+     */
+    protected $taxRate;
+
+    /**
+     * seller information.
+     *
+     * @var string
+     */
+    protected $sellerInfo;
+
+    /**
+     * due date.
+     *
+     * @var DateTime
+     */
+    protected $dueDate;
+
+    /**
+     * taxes.
+     *
+     * @var taxes
+     */
+    protected $taxes;
+
+    /**
+     * subtotal.
+     *
+     * @var float
+     */
+    protected $subTotal;
+
+    /**
+     * grand total.
+     *
+     * @var float
+     */
+    protected $grandTotal;
+
+    /**
+     * paid amount.
+     *
+     * @var float
+     */
+    protected $paidAmount;
+
+    /**
+     * due amount.
+     *
+     * @var float
+     */
+    protected $dueAmount;
+
+    /**
+     * payment terms.
+     *
+     * @var string
+     */
+    protected $paymentTerms;
+
+    /**
+     * created at.
+     *
+     * @var DateTime
+     */
+    protected $createdAt;
+
+    /**
+     * customer.
+     *
+     * @var CustomerInfoInterface
+     */
+    protected $customer;
+
+    /**
+     * invoice items.
+     *
+     * @var array|InvoiceItemInterface[]
+     */
+    protected $items;
+
+    /**
+     * invoice adjustments.
+     *
+     * @var array|InvoiceAdjustmentInterface[]
+     */
+    protected $adjustments;
+
+    /**
+     * @param string                $code
+     * @param DateTime              $date
+     * @param CurrencyInterface     $currency
+     * @param float                 $taxRate
+     * @param string                $sellerInfo
+     * @param DateTime              $dueDate
+     * @param float                 $taxes
+     * @param float                 $subTotal
+     * @param float                 $grandTotal
+     * @param float                 $paidAmount
+     * @param float                 $dueAmount
+     * @param string                $paymentTerms
+     * @param CustomerInfoInterface $customer
+     */
+    public function __construct(
+        string $code,
+        \DateTime $date,
+        CurrencyInterface $currency,
+        float $taxRate,
+        string $sellerInfo,
+        \DateTime $dueDate,
+        float $taxes,
+        float $subTotal,
+        float $grandTotal,
+        float $paidAmount,
+        float $dueAmount,
+        string $paymentTerms,
+        CustomerInfoInterface $customer
+    ) {
+        $this->code = $code;
+        $this->date = $date;
+        $this->currency = $currency;
+        $this->taxRate = $taxRate;
+        $this->sellerInfo = $sellerInfo;
+        $this->dueDate = $dueDate;
+        $this->taxes = $taxes;
+        $this->subTotal = $subTotal;
+        $this->grandTotal = $grandTotal;
+        $this->paidAmount = $paidAmount;
```

Comment:
Does this amount can have a default value `0.0` when creating invoice not yet paid without setting explicitely the paid amount to 0 ?